### ISSUE TITLE
EvalResult checkpoint_on parameter only minimizes. What about maximizable metrics?

### ISSUE THREAD

**Issue by Vichoko** [PyTorchLightning/pytorch-lightning#3861]
EvalResult `checkpoint_on` parameter, for example uses the `val_loss `metric. This metric is optimized when its minimized. But some metrics, like **accuracy** or **recall**, are optimized when maximized.

How do I tell the EvalResult logger to maximize this watched metric? When i intend to checkpoint on a maximizable metric, like the accuracy, in my case.

As stated [on the docs](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.step_result.html), there is no way to do this. 

```
def test_step(self, batch, batch_idx):
    loss = ...
    acc = ...
    result = EvalResult(checkpoint_on=acc)
    result.log('val_loss', loss)
    result.log('val_acc', acc)
    return result
```

> Parameters
> 
>         early_stop_on
>         (Optional[Tensor]) – Metric to early stop on. Should be a one element tensor if combined with default EarlyStopping. If this result is returned by validation_step(), the specified value will be averaged across all steps.
> 
>         checkpoint_on
>         (Optional[Tensor]) – Metric to checkpoint on. Should be a one element tensor if combined with default checkpoint callback. If this result is returned by validation_step(), the specified value will be averaged across all steps.
> 
>         hiddens
>         (Optional[Tensor]) –

**Comment by rohitgr7** [comment]
set `mode='max'` in [ModelCheckpoint](https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html#model-checkpointing)??

**Comment by Vichoko** [comment]
Is not clear how `ModelCheckpoint` callback on Trainer instance interacts with `EvalResult` checkpoint system as they are 2 different checkpoint systems. In my case, I tried both defining a ModelChekpoint callback with max mode and with the EvalResult checkpoint_on param, and neither worked with `val_accuracy` (the metric i'm trying to checkpoint on).

Lucklily, the workaround i figured out was to define the `error rate` metric as `1 - accuracy`, and track that metric instead as it's optimized when minimized. But for other maximized metrics isn't clear how to invert them so easily.

**Comment by williamFalcon** [comment]
EvalResult is deprecated.

Second, remember that min = -max

**Comment by Vichoko** [comment]
Thanks for the response @williamFalcon 
Did something replace EvalResult and TrainResult? What would be the best practice to log metrics and checkpointing now?

**Comment by Vichoko** [comment]
Thank you for the clarification @itsikad ^^
